### PR TITLE
Added Gphoto2 support and basic endpoints

### PR DESCRIPTION
**Pull request by theWickedWebDev**
- Removed old/unused "camera" directory
- Added new `GPhoto` class
- Split out some routes to separate files
- Added new endpoints for adding and getting camera configurations
- Backed up Thunder Client JSON requests for testing/running/etc.. endpoints
    - Telescope control, calibrate, and goto
    - Camera configuration

- Changed the following API routes to use query params:
    - `/calibrate/solar_system_object/?name=jupiter`
    - `/calibrate/by_name/?name=Polaris`
    - `/calibrate/?ra=1&dec=41d16m16.1s`
    - `/goto/by_name?name=Jupiter`
    - `/goto/solar_system_object/?name=Sun`
    - Stayed the same: /goto/0h42m45.98s/41d16m16.1s

- Added API Endpoints:
    -  `GET /api/camera/config/`
    -  `POST /api/camera/config/`
    -  `GET /api/camera/config/<_config>/` ie. `api/camera/config/iso/`
    -  `POST /api/camera/config/<_config>/<_value>/` ie. `api/camera/config/iso/800/`

### Notes
1. Want to add better exception handling (there are a few temporary placeholders currently) as well as just sending error messages out when it fails, but there are things I can do better.
    - ie. If trying to set a config value to the wrong one, I could return a list of possible values instead 

2. Need to think about how I will capture photos - perhaps in a separate thread? I need to be able to stop it if I choose to as well.

### Example Response from Get All Settings
```
{
  "/main/actions/autofocusdrive": "0",
  "/main/actions/cancelautofocus": "0",
  "/main/actions/eosmoviemode": "2",
  "/main/actions/eosremoterelease": "None",
  "/main/actions/eoszoom": "0",
  "/main/actions/eoszoomposition": "0,0",
  "/main/actions/manualfocusdrive": "None",
  "/main/actions/opcode": "0x1001,0xparam1,0xparam2",
  "/main/actions/popupflash": "2",
  "/main/actions/syncdatetime": "0",
  "/main/actions/uilock": "2",
  "/main/actions/viewfinder": "0",
  "/main/capturesettings/aeb": "off",
  "/main/capturesettings/alomode": "x3",
  "/main/capturesettings/aperture": "3.5",
  "/main/capturesettings/autoexposuremode": "Manual",
  "/main/capturesettings/bracketmode": "Unknown value 0000",
  "/main/capturesettings/drivemode": "Single",
  "/main/capturesettings/exposurecompensation": "0",
  "/main/capturesettings/focusmode": "Manual",
  "/main/capturesettings/liveviewsize": "val 0",
  "/main/capturesettings/meteringmode": "Spot",
  "/main/capturesettings/picturestyle": "Landscape",
  "/main/capturesettings/shutterspeed": "1/200",
  "/main/capturesettings/storageid": "00020001",
  "/main/imgsettings/colorspace": "sRGB",
  "/main/imgsettings/imageformat": "RAW + Large Fine JPEG",
  "/main/imgsettings/imageformatsd": "RAW + Large Fine JPEG",
  "/main/imgsettings/iso": "800",
  "/main/imgsettings/whitebalance": "Daylight",
  "/main/imgsettings/whitebalanceadjusta": "0",
  "/main/imgsettings/whitebalanceadjustb": "0",
  "/main/imgsettings/whitebalancexa": "0",
  "/main/imgsettings/whitebalancexb": "0",
  "/main/other/5001": "100",
  "/main/other/d303": "1",
  "/main/other/d402": "Canon EOS REBEL T2i",
  "/main/other/d406": "Unknown Initiator",
  "/main/other/d407": "1",
  "/main/settings/artist": "Stephen Young",
  "/main/settings/autopoweroff": "0",
  "/main/settings/capture": "0",
  "/main/settings/capturetarget": "Internal RAM",
  "/main/settings/copyright": "",
  "/main/settings/customfuncex": "c8,4,1,2c,3,101,1,0,103,1,0,10f,1,1,2,2c,3,201,1,0,202,1,0,203,1,0,3,20,2,50e,1,0,60f,1,0,4,38,4,701,1,0,704,1,0,811,1,0,80f,1,0,",
  "/main/settings/datetime": "1697743182",
  "/main/settings/depthoffield": "0",
  "/main/settings/eventmode": "0",
  "/main/settings/evfmode": "1",
  "/main/settings/flashcharged": "0",
  "/main/settings/focusinfo": "eosversion=0,size=4752x3168,size2=5184x3456,points={{-1363,0,172,129},{-819,387,172,129},{-819,-387,172,129},{0,729,117,181},{0,0,221,222},{0,-729,117,181},{819,387,172,129},{819,-387,172,129},{1363,0,172,129}},select={},unknown={10000000ffff}",
  "/main/settings/movierecordtarget": "None",
  "/main/settings/output": "Off",
  "/main/settings/ownername": "Stephen Young",
  "/main/settings/remotemode": "1",
  "/main/settings/reviewtime": "None",
  "/main/settings/strobofiring": "1",
  "/main/status/Battery Level": "100%",
  "/main/status/availableshots": "30655",
  "/main/status/batterylevel": "100%",
  "/main/status/cameramodel": "Canon EOS REBEL T2i",
  "/main/status/deviceversion": "3-1.0.9",
  "/main/status/eosmovieswitch": "0",
  "/main/status/eosserialnumber": "2123240612",
  "/main/status/lensname": "EF50mm f/1.8 II",
  "/main/status/manufacturer": "Canon Inc.",
  "/main/status/model": "2147484272",
  "/main/status/ptpversion": "256",
  "/main/status/serialnumber": "d5fb8b147c64611",
  "/main/status/shuttercounter": "47896",
  "/main/status/vendorextension": ""
}```

### CODE REVIEW

File: gphoto2/gphoto.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from gphoto2.util import getCurrentConfigValueFromCamera, setConfigValueOnCamera, getAllConfigFromCamera, setMultipleValuesOnCamera
+
+WEATHER_API_URL = 'http://wttr.in?format="%t+%w+%m+%M+%P+%z+%h+%C"'
+
+
+class GPhoto:
+    weather = {}
+    meta = {
+        "focalLength": "Not set"
+    }
```
Comment (chris-chambers) [inline gphoto2/gphoto.py]: You'll have problems with these if there is more than one instance of `GPhoto`.  For example, try:

```python
a = GPhoto()
a.weather["foo"] = "bar"

b = GPhoto()
print(b.weather["foo"]) # prints "bar", even though you never set it on `b.weather`
```

These should be initialized in `__init__` below.  It's perfectly fine in Python not to even mention a field in the class body, and just to write:

```python
def __init__(self):
    self.weather = {}
    self.meta = {
        "focalLength": "Not set"
    }
```

With those as the first mention that the fields will exist in the class.

File: handlers/main.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from flask import make_response, jsonify
+from rich import print_json
```
Comment (chris-chambers) [inline handlers/main.py:2]: Do you still need `rich`?

File: appserver.py [hunk 3]
```diff
@@ -59,100 +31,131 @@ def calib(target: tc.Target, track: bool = True):
...
+        "/api/camera/config/", methods=["GET"], view_func=camera_config_get_all)
+
+    app.add_url_rule(
+        "/api/camera/config/", methods=["POST"], view_func=camera_config_set_all)
+
+    app.add_url_rule(
+        "/api/camera/config/<_config>/", methods=["GET"], view_func=camera_config_get)
+
+    app.add_url_rule(
+        "/api/camera/config/<_config>/<_value>/", methods=["POST"], view_func=camera_config_set)
```
Comment (chris-chambers) [inline appserver.py:151]: Aw, no blueprints?

File: handlers/camera.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+from flask import request
+from gphoto2.gphoto import GPhoto
+from handlers.main import returnResponse
+
+Cam = GPhoto()
```
Comment (chris-chambers) [inline handlers/camera.py:5]: Avoiding the need to create a top-level object like this is why I did the whole `create_app` pattern.  Blueprints could let you have the best of both worlds.  It's probably not important in this case, but it definitely makes things easier to (re-)use in other ways if there isn't a forced object creation like this.  The complexity of setting it all up may not be worth it in this case.

**Review by chris-chambers — COMMENTED** [review]
Seems generally good to me.

File: appserver.py (hunk 3 above)
Comment (theWickedWebDev) [reply appserver.py:151]: Nope, lol

**Pull request merged**